Update how to manually stop and start the upgrade (#773)
harvester/docs: docs/upgrade/troubleshooting.md

:::caution

> Please do not [Start over an Upgrade](#start-over-an-upgrade) if the upgrade fails at this phase.

Reviewer: ```suggestion
Do not [restart the upgrade](#restart-an-upgrade) if the process fails at this point.
```

Reviewer: What are the consequences of restarting the upgrade at this point? What should the user do instead of restarting?

Author: Updated on https://github.com/harvester/docs/pull/773/commits/e6d39a583cbad5ba8e62ba9289174a28d16305b9, not sure the best answer, please take a new look, thanks.

## Common operations

> ### Stop the Current Upgrade
> 
> :::caution
> 
> If the current upgrade fails/is stucking at [Phase 4: Upgrade nodes](#phase-4-upgrade-nodes), check the reason first.
> 
> :::
> 
> When the current upgrade hits issues or is stucking, run below steps to stop the upgrade.

Reviewer: ```suggestion
## Stop an Ongoing Upgrade

:::caution

If an ongoing upgrade fails or becomes stuck at [Phase 4: Upgrade nodes](#phase-4-upgrade-nodes), identify the cause first.

:::

You can stop the upgrade by performing the following steps:
```

1. Log in to a control plane node.

> 1. List `Upgrade` CRs in the cluster:

Reviewer: ```suggestion
1. List the `Upgrade` CRs in the cluster.
```

    hvst-upgrade-9gmg2   10m
    ```

1. Delete the Upgrade CR

    ```
    $ kubectl delete upgrade.harvesterhci.io/hvst-upgrade-9gmg2 -n harvester-system
    ```

> 1. Resume the following Managedcharts

Reviewer 2: Though it does not harm the cluster running the script even when not doing upgrades, I think we can add more description about why they need to run this (actually, they don't need to if the previous upgrade did not get past a certain point, but I get your point).

Author: Updated, thanks.

    hvst-upgrade-9gmg2   10m
    ```

1. Delete the Upgrade CR

    ```
    $ kubectl delete upgrade.harvesterhci.io/hvst-upgrade-9gmg2 -n harvester-system
    ```

> 1. Resume the following Managedcharts

Reviewer: ```suggestion
1. Resume the paused ManagedCharts.
```

>     The upgrade pauses the managedchart to avoid data race between upgrade and others. When those managedcharts have been paused and later the upgrade is stopped, it is essential to resume them.

Reviewer: ```suggestion
  ManagedCharts are paused to avoid a data race between the upgrade and other processes. You must manually resume all paused ManagedCharts.
```

    ```
    cat > resumeallcharts.sh << 'FOE'
    resume_all_charts() {

      local patchfile="/tmp/charttmp.yaml"

      cat >"$patchfile" << 'EOF'
    spec:
      paused: false
    EOF
      echo "the to-be-patched file"
      cat "$patchfile"

      local charts="harvester harvester-crd rancher-monitoring-crd rancher-logging-crd"

      for chart in $charts; do
        echo "unapuse managedchart $chart"
        kubectl patch managedcharts.management.cattle.io $chart -n fleet-local --patch-file "$patchfile" --type merge || echo "failed, check reason"
      done

      rm "$patchfile"
    }

    resume_all_charts

    FOE

    chmod +x ./resumeallcharts.sh

    ./resumeallcharts.sh

    ```

> ### Start over an Upgrade
> 
> 1. [Stop the Current upgrade](#stop-the-current-upgrade)
> 
> 1. Click the upgrade button in the Harvester dashboard to start an upgrade again.
> 
>     If you [Customize the Version](./automatic.md#customize-the-version) manually, you might need to [create the version object](./automatic.md#prepare-the-version) again.

Reviewer: ```suggestion
### Restart an Upgrade

1. [Stop the ongoing upgrade](#stop-the-current-upgrade).

1. Click the **Upgrade** button on the Harvester UI **Dashboard** screen.

  If you [customized the version](./automatic.md#customize-the-version), you might need to [create the version object](./automatic.md#prepare-the-version) again.
```
